### PR TITLE
refactor: [DIOS-6937] Use safer async wrapper with weak_ptr object lifetimes

### PR DESCRIPTION
**Pull request by bcostdolby**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
   "dependencies": {
     "medooze-event-emitter": "^1.0.0",
     "nan": "^2.18.0",
+    "node-gyp": "^10.2.0",
```
Comment (bcostdolby) [inline package.json:35]: remove

File: src/IncomingStreamBridge.i [hunk 2]
```diff
@@ -3,15 +3,21 @@
 
 class IncomingStreamBridge : 
 	public RTMPMediaStream::Listener,
-	public RTPReceiver
+	public RTPReceiver//,
+	//public TimeServiceWrapper<IncomingStreamBridge>
+	// Cant derive from this as the loop is constructed afterwards. Need to derive from loop first then theis if we want to do this. Also TS wont exist outside the this object so safe to NOT use this pattern
```
Comment (bcostdolby) [inline src/IncomingStreamBridge.i:8]: fix

File: src/IncomingStreamBridge.i [hunk 3]
```diff
@@ -3,15 +3,21 @@
...
 {
 private:
 	static constexpr size_t BaseVideoSSRC = 2;
+
+// @todo owns the loop so ok
```
Comment (bcostdolby) [inline src/IncomingStreamBridge.i:13]: fix

File: src/MediaFrameListenerBridge.i [hunk 4]
```diff
@@ -16,7 +16,10 @@ struct MediaFrameListenerBridge :
 	public MediaFrameListener,
 	public MediaFrameProducer
 {
-	MediaFrameListenerBridge(TimeService& timeService, int ssrc);
+private:
+	MediaFrameListenerBridge();
+public:
+	//MediaFrameListenerBridge(TimeService& timeService, int ssrc);
```
Comment (bcostdolby) [inline src/MediaFrameListenerBridge.i:22]: fix

**Review by harryz2000 — APPROVED** [review]
lgtm

**Pull request merged**